bluesky/plans.py

```diff
@@ -964,10 +964,27 @@ def relative_inner_product_scan(detectors, num, *args, per_step=None, md=None):
     :func:`bluesky.plans.inner_product_scan`
     :func:`bluesky.plans.scan_nd`
     """
-    _md = {'plan_name': 'relative_inner_product_scan'}
+    _md = {'plan_name': 'relative_inner_product_scan',
+           }
     _md.update(md or {})
     motors = [motor for motor, start, stop in partition(3, args)]
 
+    # Default should assume the first motor is what to plot
+    # If another motor is desired, override with separate hints
+    if hasattr(motors[0], 'hints') and 'fields' in motors[0].hints:
```

Comment:
Do we need this conditional?  Could we do this as

```python
for motor in motors:
    fields.extend(getattr(motor, 'hints', {}).get('fields', []))
```

?

Reply:
yes that's much more elegant thanks! made a few extra changes